pomegranate/NaiveBayes.pyx

```diff
@@ -58,23 +58,40 @@ cdef class NaiveBayes( object ):
            [-0.26751248, -1.4493653 ],
            [-1.09861229, -0.40546511]])
     """
-
+    
     cdef int initialized
     cdef public object models
     cdef void** models_ptr
     cdef numpy.ndarray summaries
     cdef public numpy.ndarray weights
     cdef double* weights_ptr
+    # dimension of inputs, 0 is dimensionless, -1 if uninitialized
+    cdef public int d
 
     def __init__( self, models=None, weights=None ):
         if not callable(models) and not isinstance(models, list): 
             raise ValueError("must either give initial models or constructor")
 
         self.summaries = None
-        self.initialized = False
-
-        if not callable(models):
-            self.initialized = True
+        self.d = -1
+        
+        if type(models) is list:
+            if not callable( models[0] ):
+                if isinstance( models[0], HiddenMarkovModel ):
+                    self.d = 0
+                else:
+                    self.d = models[0].d
+            else:
+                raise TypeError("must have initialized models in list")
+
+            for model in models:
+                if callable(model):
+                    raise TypeError("must have initialized models in list")
+                elif self.d == 0 and not isinstance( model, HiddenMarkovModel ):
```

Comment:
This isn't necessarily true either. You should be able to compare a HMM to a Bayes net or GMM if they have the same dimensionality, aka a sequence of length 3 in 1 dimension can be considered the same as a data point of dimension 3, it's just that the HMM imposes structure whereas the others assume independence.